crates/loro-internal/src/undo.rs

```diff
@@ -415,10 +421,34 @@ impl UndoManagerInner {
             last_popped_selection: None,
             on_pop: None,
             on_push: None,
+            group_active: false,
+            group_start_counter: None,
+            group_affected_cids: Default::default(),
         }
     }
 
+    /// Resets the current group to have no active group.
+    fn reset_group(&mut self) {
+        self.group_active = false;
+        self.group_start_counter = None;
+        self.group_affected_cids.clear();
+    }
+
+    /// Returns true if a given container diff is disjoint with the current group.
+    /// They are disjoint if they have no ovrlap in changed container ids.
+    fn is_disjoint_with_group(&self, diff: &[&ContainerDiff]) -> bool {
+        let incoming_cids: FxHashSet<ContainerID> = diff.iter().map(|d| d.id.clone()).collect();
```

Comment:
We can early return here to avoid unnecessary calculations

```rust
if !self.group_active {
    return false
}

```

Reply:
fixed